playwright/test/search-works.test.ts

```diff
@@ -1,336 +1,282 @@
+  expect(originalSearchUrl.pathname).toEqual(backToSearchResultsUrl.pathname);
 
-test.describe.configure({ mode: 'parallel' });
+  originalSearchUrl.searchParams.sort();
+  backToSearchResultsUrl.searchParams.sort();
 
-test.describe('Scenario 1: The person is looking for an archive', () => {
-  test('the work should be browsable to from the search results', async ({
-    page,
-    context,
-  }) => {
-    await newWorksSearch(context, page);
-    await searchFor('Persian', page);
-    await openDropdown('Formats', page);
-    await selectCheckbox('Archives and manuscripts', page);
-    await navigateToNextPage(page);
-
-    expectSearchParam('workType', 'h', page);
-
-    await navigateToResult(3, page);
-  });
-
-  test('and the user can get back to their original search results', async ({
-    page,
-    context,
-  }) => {
-    // Open the search page, search for a term, apply a filter, navigate away from
-    // the first page.
-    await newWorksSearch(context, page);
-    await searchFor('Persian', page);
-    await openDropdown('Formats', page);
-    await selectCheckbox('Archives and manuscripts', page);
-    await navigateToNextPage(page);
-
-    // Save the URL of the current search page, which will be something like
-    // https://www-stage.wellcomecollection.org/search/works?query=Persian&workType=h&page=2
-    const originalSearchUrl = new URL(page.url());
-
-    // Now go to the third result on the page, and look for "Back to search results".
-    await navigateToResult(3, page);
-
-    const backToSearchResults = await page.$(
-      'a:has-text("Back to search results")'
-    );
-    expect(backToSearchResults).toBeTruthy();
-
-    // Note: we expect the URLs on the "Back to search results" link to be relative,
-    // but the URL class only takes absolute URLs.
-    //
-    // Check this is a relative URL, then construct an absolute URL we can compare.
-    const backToSearchResultsUrlString =
-      (await backToSearchResults?.getAttribute('href')) as string;
-    expect(backToSearchResultsUrlString.startsWith('/')).toBe(true);
-
-    const backToSearchResultsUrl = new URL(
-      `${process.env.PLAYWRIGHT_BASE_URL}${backToSearchResultsUrlString}`
-    );
-
-    // Now compare the URLs.  Note that the query parameters may be in a different order,
-    // but they're still equivalent for our purposes, e.g.
-    //
-    //      /search/works?query=Persian&workType=h&page=2 and
-    //      /search/works?query=Persian&page=2&workType=h
-    //
-    // are both totally fine.  Sorting them first makes them easier to compare.
-    expect(originalSearchUrl.pathname).toEqual(backToSearchResultsUrl.pathname);
-
-    originalSearchUrl.searchParams.sort();
-    backToSearchResultsUrl.searchParams.sort();
-
-    expect(originalSearchUrl.searchParams).toEqual(
-      backToSearchResultsUrl.searchParams
-    );
-  });
+  expect(originalSearchUrl.searchParams).toEqual(
+    backToSearchResultsUrl.searchParams
+  );
```

Comment:
What if we just click the "back to search results" link and make sure that we go back to `originalSearchURL`? This seems like one of the legitimate use cases for `waitForURL`

Reply:
> // Now compare the URLs. Note that the query parameters may be in a different order

This does happen though, so will still need to reorder them I think.

Reply:
but we can do that in `waitForURL` right? the core idea being that we don't need to mess around with inspecting the link contents

Reply:
Indeed! Trialled a first version here https://github.com/wellcomecollection/wellcomecollection.org/pull/10414/files#diff-b71f84b7d77f88b09fa5d1c65a2106819191ee085f52cc31cd67e8948fb9b7d1R51